plugins/repository-azure/src/main/java/org/elasticsearch/repositories/azure/AzureStorageService.java

```diff
@@ -187,72 +188,61 @@ public boolean blobExists(String account, String container, String blob) throws
         });
     }
 
-    public void deleteBlob(String account, String container, String blob) throws URISyntaxException, StorageException {
+    public void deleteBlobsIgnoringIfNotExists(String account, String container, Collection<String> blobs)
+            throws URISyntaxException, StorageException {
         final Tuple<CloudBlobClient, Supplier<OperationContext>> client = client(account);
         // Container name must be lower case.
         final CloudBlobContainer blobContainer = client.v1().getContainerReference(container);
-        logger.trace(() -> new ParameterizedMessage("delete blob for container [{}], blob [{}]", container, blob));
-        SocketAccess.doPrivilegedVoidException(() -> {
-            final CloudBlockBlob azureBlob = blobContainer.getBlockBlobReference(blob);
-            logger.trace(() -> new ParameterizedMessage("container [{}]: blob [{}] found. removing.", container, blob));
-            azureBlob.delete(DeleteSnapshotsOption.NONE, null, null, client.v2().get());
-        });
+        final Iterator<String> blobIterator = blobs.iterator();
+        int currentBatchSize = 0;
+        while (blobIterator.hasNext()) {
+            final BlobDeleteBatchOperation batchDeleteOp = new BlobDeleteBatchOperation();
+            do {
+                batchDeleteOp.addSubOperation(blobContainer.getBlockBlobReference(blobIterator.next()),
+                    DeleteSnapshotsOption.NONE, null, null);
+                ++currentBatchSize;
+            } while (blobIterator.hasNext() && currentBatchSize < Constants.BATCH_MAX_REQUESTS);
+            currentBatchSize = 0;
+            logger.trace(() -> new ParameterizedMessage("delete blobs for container [{}], blob [{}]", container, blobs));
```

Comment:
In case of more than 256 blobs to delete, this will log the same message multiple times right?

Reply:
Right :) moved the log statement to the beginning of the method now